VideoLocker/src/main/java/org/edx/mobile/task/GetThreadListTask.java

```diff
@@ -33,31 +32,13 @@ public GetThreadListTask(Context context,
         this.pagination = pagination;
     }
 
-
-
-    public TopicThreads call( ) throws Exception{
+    public TopicThreads call() throws Exception {
         try {
-
-            if(courseId!=null){
-
-                String view;
-                if (filter == DiscussionPostsFilter.UNREAD) view = "unread";
-                else if (filter == DiscussionPostsFilter.UNANSWERED) view = "unanswered";
-                else view = "";
-
-                String order;
-                if (orderBy == DiscussionPostsSort.LAST_ACTIVITY_AT) order = "last_activity_at";
-                else if (orderBy == DiscussionPostsSort.VOTE_COUNT) order = "vote_count";
-                else order = "";
-
+            if (courseId != null) {
                 int pageSize = pagination.pageSize();
                 int page = pagination.numOfPagesLoaded() + 1;
-
                 return environment.getDiscussionAPI().getThreadList(courseId, topicIds,
-                        view,
-                        order,
-                        pageSize,
-                        page);
+                        filter.getQueryParamValue(), orderBy.getQueryParamValue(), pageSize, page);
             }
         } catch (Exception ex) {
             handle(ex);
```

Comment:
Instead of catching generic `Exception`, we should probably only have `RetroHttpException` caught here. I don't think there can be any other exception thrown in this block. We should probably have the logic refactored in `Task` at some point where it automatically calls `handle()` from `onException()`, but I guess we can leave it for now.